Fails to build on !Linux since 22845a97

Heya,

in 22845a97 you introduced a new file: driver/driver_config.h.in which generates driver/driver_config.h. The later is then used by the userspace code to get a few variables.

However, driver/driver_config.h is only built, when the driver directory is built. Which is not the case on non-Linux systems (I tried on Debian/kFreeBSD and Debian/Hurd).

The build then fails with the following error

    userspace/libscap/scap.c:34:40: fatal error: ../../driver/driver_config.h: No such file or directory

Putting `add_subdirectory(driver)` outside of `if(CMAKE_SYSTEM_NAME MATCHES "Linux")` worked for my testbuilds (they have `BUILD_DRIVER=OFF` anyways) but I don't really like that solution yet.

--- a/userspace/libscap/scap.c
+++ b/userspace/libscap/scap.c
@@ -30,9 +30,9 @@ along with sysdig.  If not, see <http://www.gnu.org/licenses/>.
 #endif // _WIN32
 
 #include "scap.h"
-#ifndef _WIN32
+#ifdef HAS_CAPTURE
 #include "../../driver/driver_config.h"
-#endif // _WIN32
+#endif // HAS_CAPTURE
 #include "../../driver/ppm_ringbuffer.h"
 #include "scap_savefile.h"
 #include "scap-int.h"

--- a/userspace/sysdig/sysdig.h
+++ b/userspace/sysdig/sysdig.h
@@ -19,9 +19,9 @@ along with sysdig.  If not, see <http://www.gnu.org/licenses/>.
 #pragma once
 
 #include <config_sysdig.h>
-#ifndef _WIN32
+#ifdef HAS_CAPTURE
 #include "../../driver/driver_config.h"
-#endif // _WIN32
+#endif // HAS_CAPTURE
 
 //
 // ASSERT implementation

